### PR TITLE
Assert conditional SEG names for AVI configmap in bootstrap clusters.

### PR DESCRIPTION
**Pull request by HanFa**
### What this PR does / why we need it

What it does: 
1. Refactor some of the repeated setup logics into `BeforeEach` function.
2. Add 2 unit tests that assert correct conditionally SEG name rendering inside bootstrap cluster

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note
Assert conditional SEG names for AVI configmap in bootstrap clusters
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by lubronzhan — APPROVED** [review]
lgtm

**Comment by HanFa** [comment]
could you fix the linting for this markdown? thanks @tylerschultz

File: pkg/v1/providers/ytt/04_user_customizations/README.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # User Customizations
 
-ytt traverses directories in alphabetical order and overwrites duplicate settings. You can create template files here to contain configurations that take precedence over any in lower-numbered ytt subdirectories.
\ No newline at end of file
+ytt traverses directories in alphabetical order and overwrites duplicate settings. You can create template files here to contain configurations that take precedence over any in lower-numbered ytt subdirectories.
```
Comment (vuil) [inline pkg/v1/providers/ytt/04_user_customizations/README.md:3]: *wiping my eyes* where is the diff :-) ?!

**Review by vuil — APPROVED** [review]
lgtm, thanks!

File: pkg/v1/providers/ytt/04_user_customizations/README.md (hunk 1 above)
Comment (HanFa) [reply pkg/v1/providers/ytt/04_user_customizations/README.md:3]: oh the CI linting was complaining there is no newline character. So I added a `\n`

**Pull request merged**